extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -760,7 +760,7 @@ private void enhanceEntities(final JpaEntitiesBuildItem domainObjects,
             try {
                 byte[] bytes = IoUtil.readClassAsBytes(HibernateOrmProcessor.class.getClassLoader(), className);
                 byte[] enhanced = hibernateEntityEnhancer.enhance(className, bytes);
-                additionalClasses.produce(new GeneratedClassBuildItem(true, className, enhanced != null ? enhanced : bytes));
+                additionalClasses.produce(new GeneratedClassBuildItem(false, className, enhanced != null ? enhanced : bytes));
```

Comment:
We are enhancing application classes and I don't understand why the enhanced classes should not be considered application classes?

Reply:
These are AdditionalJpaModelBuildItem classes, that are for entities defined in extensions. If we generate them in the application class loader then you end up with two copies, one in the runtime class loader and the original unmodified one in the base runtime class loader. Normally this is ok, unless a class in the base runtime CL attempts to use the entity, in which case you get problems.